### ISSUE TITLE
build error narrowing conversion of ‘-9’ from ‘int’ to ‘std::size_t'

### ISSUE THREAD

**Issue by DSLituiev** [xiangzhou/GEMMA#23]
I get following build error with `gcc (GCC) 5.2.1 20150902`:

```
src/io.cpp:2056:61: error: narrowing conversion of ‘-9’ from ‘int’ to ‘std::size_t {aka long unsigned int}’ inside { } [-Wnarrowing]
    SNPINFO sInfo={"-9", rs, -9, -9, minor, major, -9, -9, -9};
```

**Comment by xiangzhu** [comment]
@xiangzhou I get the same build error with `gcc-5`; and I also get the same error using `g++ (Ubuntu 6.2.0-3ubuntu11~16.04) 6.2.0 20160901`.

Here is the error from `gcc-6`.
```
src/io.cpp: In function ‘bool ReadFile_bgen(const string&, const std::set<std::__cxx11::basic_string<char> >&, const gsl_matrix*, std::vector<int>&, std::vector<int>&, std::vector<SNPINFO>&, const double&, const double&, const double&, const double&, std::size_t&)’:
src/io.cpp:2165:73: error: narrowing conversion of ‘-9’ from ‘int’ to ‘std::size_t {aka long unsigned int}’ inside { } [-Wnarrowing]
     SNPINFO sInfo={"-9", rs, -9, -9, minor, major, -9, -9, (long int) -9};
                                                                         ^
Makefile:108: recipe for target 'src/io.o' failed
make: *** [src/io.o] Error 1
```

**Comment by wanyuac** [comment]
The same to me on ubuntu:

src/io.cpp: In function ‘bool ReadFile_bgen(const string&, const std::set<std::__cxx11::basic_string<char> >&, const gsl_matrix*, std::vector<int>&, std::vector<int>&, std::vector<SNPINFO>&, const double&, const double&, const double&, const double&, std::size_t&)’:
src/io.cpp:2165:73: error: narrowing conversion of ‘-9’ from ‘int’ to ‘std::size_t {aka long unsigned int}’ inside { } [-Wnarrowing]
     SNPINFO sInfo={"-9", rs, -9, -9, minor, major, -9, -9, (long int) -9};
                                                                         ^
Makefile:108: recipe for target 'src/io.o' failed
make: *** [src/io.o] Error 1

**Comment by wanyuac** [comment]
It is a problem of the compiler we have used. As one of my colleagues suspected, I finally got GEMMA compiled successfully using gcc-4.9.3. It seems that the code "SNPINFO sInfo={"-9", rs, -9, -9, minor, major, -9, -9, (long int) -9};" is not compatible with gcc-5+. @xiangzhu @DSLituiev 

There is another way to get round of this problem using gcc-5. We can change the class definition:

``
class SNPINFO {
public:
	...
	size_t n_miss;
	...
};
``

into

``
class SNPINFO {
public:
	...
	**ssize_t** n_miss;
	...
};
``

However, I would not recommend this method because I am not sure if the substitution of data types would affect results.

**Comment by echubaty** [comment]
@wanyuac Thank you, your fork of this repo worked for me. The only modification I had to make for my environment was changing CPP to g++-4.9 since I now have multiple versions installed.

**Comment by pcarbo** [comment]
@wanyuac @xiangzhu @echubaty @DSLituiev Can one of you please check to see if this issue is now resolved? See also Issue #24, which reported the same problem (although using a different compiler).

**Comment by pjotrp** [comment]
Guix version compiles fine with latest gcc compilers.
